libs/gl-client/src/signer/mod.rs

```diff
@@ -247,19 +266,89 @@ impl Signer {
                 }
 
                 pk.verify(&data, &r.signature)
+                    .map_err(|e| anyhow!("signature verification failed: {}", e))?;
+
+                self.verify_rune(r.clone())
                     .map(|_| r)
-                    .map_err(|e| anyhow!("signature verification failed: {}", e))
+                    .map_err(|e| anyhow!("rune verification failed: {}", e))
             })
             .collect()
     }
 
+    /// Verifies that the public key of the request and the signers rune version
+    /// match the corresponding restrictions of the rune.
+    fn verify_rune(&self, request: crate::pb::PendingRequest) -> Result<(), anyhow::Error> {
+        let rune64 = general_purpose::URL_SAFE.encode(request.rune);
+        let rune = Rune::from_base64(&rune64)?;
+
+        // A valid gl-rune must contain a pubkey field as this  is bound to the
+        // signer. Against the rules of runes we do not accept a rune that has
+        // no restriction on a public key.
+        if !rune.to_string().contains("pubkey=") {
+            return Err(anyhow!("rune is missing pubkey field"));
+        }
+
+        let mut checks: HashMap<String, String> = HashMap::new();
+        checks.insert("pubkey".to_string(), hex::encode(request.pubkey));
+
+        // Runes only check on the version if the unique id field is set. The id
+        // and the version are part of the empty field.
+        if let Some(device_id) = rune.get_id() {
+            checks.insert("".to_string(), format!("{}-{}", device_id, RUNE_VERSION));
+        }
+
+        // Check that the request points to `cln.Node`.
+        let mut parts = request.uri.split('/');
+        parts.next();
+        match parts.next() {
+            Some(service) => {
+                // Fixme: We skip this for now as we also have the
```

Comment:
What are we skipping and how would stricter validation look?

Reply:
Oh the comment seems to be off, actually we DO check the grpc service that was used to invoke the request.6ac1eb973ff4b60ce06436cb4fb2c171a9903bd2